### PR TITLE
Audio : allow XAudio2 to run its own thread

### PR DESCRIPTION
**Pull request by raven02**
It seems to fix the issues . Please test it again especially with PPU LLVM where easily trigger the AddData() failed.

### CODE REVIEW

File: rpcs3/Emu/Audio/XAudio2/XAudio27Thread.cpp [hunk 1]
```diff
@@ -30,6 +30,7 @@ void XAudio2Thread::xa27_init()
 	if (FAILED(hr))
 	{
 		LOG_ERROR(GENERAL, "XAudio2Thread : XAudio2Create() failed(0x%08x)", (u32)hr);
+		CoUninitialize();
```
Comment (Nekotekina) [inline rpcs3/Emu/Audio/XAudio2/XAudio27Thread.cpp]: It's already called in the destructor

**Comment by raven02** [comment]
@kd-11 , did u still hit the error using this build when thread stall ?

**Comment by raven02** [comment]
@danilaml / @Kristiqn5 / @Nezarn . Please test it when build done

File: rpcs3/Emu/Audio/XAudio2/XAudio27Thread.cpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ void XAudio2Thread::xa27_init()
 	{
 		LOG_ERROR(GENERAL, "XAudio2Thread : CreateMasteringVoice() failed(0x%08x)", (u32)hr);
 		s_tls_xaudio2_instance->Release();
+		CoUninitialize();
```
Comment (Nekotekina) [inline rpcs3/Emu/Audio/XAudio2/XAudio27Thread.cpp]: ...

File: rpcs3/Emu/Audio/XAudio2/XAudio28Thread.cpp [hunk 3]
```diff
@@ -15,11 +15,19 @@ static thread_local IXAudio2MasteringVoice* s_tls_master_voice{};
 static thread_local IXAudio2SourceVoice* s_tls_source_voice{};
 
 void XAudio2Thread::xa28_init(void* module)
-{
+{
```
Comment (GhettoGirl) [inline rpcs3/Emu/Audio/XAudio2/XAudio28Thread.cpp]: unnecessary whitespace

**Comment by kd-11** [comment]
Haven't been able to reproduce crashes, although it usually isn't 100% reproducible. We'll need a larger number of testers for this one since threading issues are hard to reproduce.

**Comment by Nezarn** [comment]
@raven02 works fine on my system. (i even tried doing what was mentioned in #1973 , wasn't able to reproduce)

**Comment by Nekotekina** [comment]
Just copy from existing XAudio27Thread.cpp implementation

**Comment by danilaml** [comment]
Btw, might be worth reading (if you haven't already): https://blogs.msdn.microsoft.com/chuckw/2012/04/02/xaudio2-and-windows-8/

**Comment by raven02** [comment]
I'll try to merge it atm with testers positive result .

**Comment by Nekotekina** [comment]
Damn could you at least implement it correctly?

**Comment by raven02** [comment]
I see others didn't get this error anymore so merge it. I'll revert it
